### PR TITLE
8490: delete API key endpoint

### PR DESCRIPTION
**Pull request by mkalish**
This PR adds an endpoint to delete public keys from an organization

Test Steps:
1. Either add a key to an organization or can use an existing key.
<details>
<summary> cURL</summary>

```sh
curl --location --globoff '{{url}}/settings/organizations/simple_report/public-keys?scope=simple_report.*.report&kid=simple_report.default1' \
--header 'Authorization: Bearer eyJhbGciOiJIUzM4NCJ9.eyJleHAiOjE2Nzg0NjM3NDAsInNjb3BlIjoic2ltcGxlX3JlcG9ydC4qLnJlcG9ydCIsInN1YiI6InNpbXBsZV9yZXBvcnQuKi5yZXBvcnRfMTU0OTZlZjUtZTY3YS00MDkxLTg3Y2EtMGUzMGU1ZWQ0YjZkIn0.9t8gD2bQan6hkM7MmAy-TxoPdtAAEFkodPFwqV-8qzc6N2v2wkY_dsSHJTUrgOJw' \
--header 'Content-Type: text/plain' \
--data '-----BEGIN PUBLIC KEY-----
MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAu/iNl3p3b0G4a5AbhycX
0gnzszqSjMP7NJygno4UfF8HA+jLKqKXbG93ywyvbau+Rm1J7wbVb5hYWx3jvXt2
7LfQea1XgMF67O2Xo7bJ7U5NSceVVrtAA/IIq5sqolrxcVQ5LbGkuVqVTKwMtgs2
pe+sj6GcgdjhRnxLCN0Ex4xB7RtsieIvqwqEi4ds+sbLAs9fHMumcGzvyDLQpNGW
avOMqpXM1WqNauzOn4mgECu4BJuHPu7gXIPAkrVwLRF0LcSs7Js37uWtg04FxTz8
6sv0kYwqYXDJNcrCd+3S5cbQTGvdLAbUvvZ3vassfrxbQGFKJEt51iiHW7U/CaX8
ewIDAQAB
-----END PUBLIC KEY-----'
```
</details>
2. Check that the key is there

```sh
curl --location 'http://localhost:7071/api/settings/organizations/simple_report/public-keys' \
--header 'Authorization: Bearer dummy'
```
3. Invoke the delete endpoint and verify that the response does not contain the key

```sh
curl --location --request DELETE 'http://localhost:7071/api/settings/organizations/simple_report/public-keys/simple_report.%2A.report/simple_report%2Edefault1' \
--header 'Authorization: Bearer dummy'
```

4. Check that the key is now gone

```sh
curl --location 'http://localhost:7071/api/settings/organizations/simple_report/public-keys' \
--header 'Authorization: Bearer dummy'
```
Screenshot 2023-04-13 at 12 24 46 PM

## Changes
- Adds an endpoint for deleting keys for the scope org.*.report

### Testing
- [x] Tested locally?
- [ ] Ran `./prime test` or `./gradlew testSmoke` against local Docker ReportStream container?
- [ ] (For Changes to /frontend-react/...) Ran `npm run lint:write`? 
- [x] Added tests?

### Process
- [ ] Are there licensing issues with any new dependencies introduced?
- [ ] Includes a summary of what a code reviewer should test/verify?
- [ ] Updated the release notes?
- [ ] Database changes are submitted as a separate PR?
- [ ] DevOps team has been notified if PR requires ops support?

## Linked Issues
- Fixes #issue

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #issue

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?

### CODE REVIEW

File: prime-router/src/main/kotlin/cli/tests/AuthTests.kt [hunk 1]
```diff
@@ -551,6 +552,71 @@ class Server2ServerAuthTests : CoolTest() {
...
+        )
+
+        if (httpStatusDeleteKey == 200) {
+            good("Successfully removed key from ${org.name}")
+        } else {
+            bad("Failed to remove key from ${org.name}, response was $deleteKeyResponse")
+            passed = false
+        }
+
         return passed
```
Comment (arnejduranovic) [inline prime-router/src/main/kotlin/cli/tests/AuthTests.kt:620]: Should we do a call to get keys and make sure it is gone and we did not get an erroneous 200?

Comment (mkalish) [reply prime-router/src/main/kotlin/cli/tests/AuthTests.kt:620]: Yep, that's a good call

**Review by arnejduranovic — CHANGES_REQUESTED** [review]
Since we added `addkey` CLI command, I think we should also add `removekey` command

**Comment by arnejduranovic** [comment]
I didn't notice any violations, but let's double-check to make sure we are conforming to this: https://github.com/CDCgov/prime-reportstream/blob/master/prime-router/docs/proposals/0019-website-api-payload-structure.md

**Comment by mkalish** [comment]
Back to you @arnejduranovic.  I think the webex team is waiting on this

File: prime-router/docs/api/api-keys.yml [hunk 2]
```diff
@@ -8,6 +8,41 @@ info:
...
+            type: string
+          required: true
+          name: scope
+          description: The scope the key to be deleted is associated with
+        - in: path
+          schema:
+            type: string
+          required: true
+          name: kid
+          description: The unique identifier for the key to be delete
```
Comment (arnejduranovic) [inline prime-router/docs/api/api-keys.yml:34]: for the key to be `deleted`

**Review by arnejduranovic — CHANGES_REQUESTED** [review]

File: prime-router/docs/api/api-keys.yml (hunk 2 above)
Comment (mkalish) [inline prime-router/docs/api/api-keys.yml:34]:
```suggestion
          description: The unique identifier of the key to be deleted
```

Comment (mkalish) [reply prime-router/docs/api/api-keys.yml]: @arnejduranovic Thanks fixed

**Review by arnejduranovic — APPROVED** [review]
Tested via command line tools, looks good!

**Pull request merged**